openmc/stats/univariate.py

```diff
@@ -947,6 +1104,27 @@ def distribution(self, distribution):
                       Iterable, Univariate)
         self._distribution = distribution
 
+    def cdf(self):
+        return np.insert(np.cumsum(self.probability), 0, 0.0)
+
+    def sample(self, n_samples=1, seed=None):
+        np.random.seed(seed)
+        xi = np.random.rand(n_samples)
+        idx = np.zeros_like(xi, dtype=int)
+        for i, c in enumerate(self.cdf()[:-1]):
+            idx[xi >= c] = i
```

Comment:
Similar to for the Discrete class, I think you could use np.random.choice here to simplify the logic a bit

Reply:
Definitely! Really handle function. Gonna try to remember that one.